### PR TITLE
Fix escaped characters in translations

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/deltachat/deltachat-desktop/issues/544, closes https://github.com/deltachat/deltachat-desktop/issues/515

screenshot from 2019-01-09 15-28-28
screenshot from 2019-01-09 15-28-53

### CODE REVIEW

**Comment by ralphtheninja** [comment]
A tonne of erroneous translations. For now I have adjusted the xml resources as well, but we should probably tweak the conversion script later to handle all the edge cases (some translators add xml tags _inside_ the resources, such as `<br/>` and `<em>foobar</em>`).

File: _locales/tr.xml [hunk 1]
```diff
@@ -236,13 +236,13 @@
...
     <string name="welcome_intro2_message">Hiç Delta Chat sunucusu yoktur. Uygulama, yalnızca posta sunucunuzu kullanır.</string>
     <string name="welcome_intro3_headline">Hızlı</string>
-    <string name="welcome_intro3_message">İletileri saniyeler içinde gönderin.<br/>Hızlı arabirim.</string>
+    <string name="welcome_intro3_message">İletileri saniyeler içinde gönderin.\nHızlı arabirim.</string>
     <string name="welcome_intro4_headline">Güçlü</string>
     <string name="welcome_intro4_message">Sınırsız sohbetler, görüntüler, videolar, sesli iletiler ve daha fazlası. Çoklu istemci yetenekli.</string>
     <!-- Translators: "free" as in freedom here, not "free" as in "free beer" -->
     <string name="welcome_intro5_headline">Özgür</string>
     <!-- Translators: "free" as in freedom here, not "free" as in "free beer" -->
-    <string name="welcome_intro5_message">Delta Chat sonsuza dek özgürdür.<br/>Açık kaynaklı. Reklâm yok. Abonelik yok. Satıcı kilidi yok.</string>
```
Comment (ralphtheninja) [inline _locales/tr.xml]: This doesn't have a new line in the original

File: test/unit/index.js [hunk 2]
```diff
@@ -14,20 +14,48 @@ test('that translation files are valid json', t => {
       jsonFiles.map(name),
       'each xml file has a corresponding json file'
     )
+    const testFile = file => {
+      let json = null
+      try {
+        json = require(file)
+      } catch (e) {
+        console.error(e.message)
+        return false
```
Comment (ralphtheninja) [inline test/unit/index.js]: First tests makes sure the file is valid json.

File: test/unit/index.js [hunk 3]
```diff
@@ -14,20 +14,48 @@ test('that translation files are valid json', t => {
...
+      }
+      return Object.keys(json).every(k1 => {
```
Comment (ralphtheninja) [inline test/unit/index.js]: For every resource ..

File: test/unit/index.js [hunk 4]
```diff
@@ -14,20 +14,48 @@ test('that translation files are valid json', t => {
...
+        const v1 = json[k1]
+        return Object.keys(v1).every(k2 => {
```
Comment (ralphtheninja) [inline test/unit/index.js]: .. every key of that resource (e.g. `"message"` etc) ..

File: test/unit/index.js [hunk 5]
```diff
@@ -14,20 +14,48 @@ test('that translation files are valid json', t => {
...
+          const v2 = v1[k2]
+          if (typeof v2 !== 'string') {
+            console.error(
+              `> ${JSON.stringify(v2)} not a string (${file} -> ${k1} -> ${k2})`
+            )
+            return false
+          }
+
```
Comment (ralphtheninja) [inline test/unit/index.js]: It should be a string ..

File: test/unit/index.js [hunk 6]
```diff
@@ -14,20 +14,48 @@ test('that translation files are valid json', t => {
...
+              console.error(`> ${JSON.stringify(v2)} contains ${str} (${file} -> ${k1} -> ${k2}) (index: ${match.index})`)
+              return false
+            }
+            return true
+          }
+
+          if (!testString('\\\\n')) return false
+          if (!testString("\\\\'")) return false
+          if (!testString('\\\\\\"')) return false
+          if (!testString('\\\\')) return false
```
Comment (ralphtheninja) [inline test/unit/index.js]: And it shouldn't contain these.

Comment (ralphtheninja) [reply test/unit/index.js]: Could probably refactor this into one `return` statement. Up to you guys.

**Comment by r10s** [comment]
> It would be nice to merge this today. It kills two bugs.

i just pulled the translations from transifex - are these already the changes from this pr?

just asking because it is a bit dangerous to push sources to transifex as this will always overwrite all work of the translators (transifex is not line bases afaik) since the last pull (16 hours ago in this case).
but probably you know this best :)

for android, therefore, i never edit the strings-LANG.xml files locally but do this always in the web-frontend.

**Comment by ralphtheninja** [comment]
@r10s I didn't push these to Transifex. I edited them on Transifex.

File: test/unit/index.js (hunk 6 above)
Comment (Jikstra) [reply test/unit/index.js]: It's fine with multiple returns to me

**Pull request merged**